pkg/virt-handler/rest/lifecycle.go

```diff
@@ -182,6 +186,36 @@ func (lh *LifecycleHandler) UnfreezeHandler(request *restful.Request, response *
 	response.WriteHeader(http.StatusAccepted)
 }
 
+func (lh *LifecycleHandler) SoftRebootHandler(request *restful.Request, response *restful.Response) {
+	vmi, code, err := getVMI(request, lh.vmiInformer)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to retrieve VMI")
+		response.WriteError(code, err)
+		return
+	}
+
+	sockFile, err := cmdclient.FindSocketOnHost(vmi)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to detect cmd client")
+		response.WriteError(http.StatusInternalServerError, err)
+	}
+	client, err := cmdclient.NewClient(sockFile)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to connect cmd client")
+		response.WriteError(http.StatusInternalServerError, err)
+	}
+
+	err = client.SoftRebootVirtualMachine(vmi)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to soft reboot VMI")
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
```

Comment:
Everywhere that `WriteError` occurs needs to return early. I think this can technically cause virt-handler to crash otherwise

For example, if socketFile doesn't exist, then NewClient will return nil for the client, and then the client is dereferenced `err = client.SoftRebootVirtualMachine(vmi)`

I know your code is modeled after the pause logic. Whoever wrote that got the same problem wrong. I'll fix that in a separate pr.

Reply:
I fix the other areas in  #6841. This new code for reboot needs to follow the same pattern in #6841